ext/objspace/objspace_dump.c

```diff
@@ -323,9 +333,17 @@ dump_append_string_content(struct dump_config *dc, VALUE obj)
         dump_append_sizet(dc, rb_str_capacity(obj));
     }
 
-    if (is_ascii_string(obj)) {
-        dump_append(dc, ", \"value\":");
-        dump_append_string_value(dc, obj);
+    if (RSTRING_LEN(obj)) {
+        int cr = ENC_CODERANGE(obj);
+        if (cr == RUBY_ENC_CODERANGE_UNKNOWN) {
+            if (dump_string_ascii_only(RSTRING_PTR(obj), RSTRING_LEN(obj))) {
+                cr = RUBY_ENC_CODERANGE_7BIT;
```

Comment:
This might not be accurate. You could have a non-ASCII-compatible encoding that happens to consists of only bytes in the ASCII range. In that case, you'd get a different value than `rb_enc_str_coderange` and, by extension, `rb_str_is_ascii_only_p`.

Reply:
Good catch, I need to also check for `ascii_compatible`.